homework-3/src/components/Forms/FormSignIn.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { Formik } from 'formik';
+import TitleForm from '../common/Titles/StylTitleForm';
+import FormInput from '../FormInput';
+import FormSign from '../common/partsOfTheForm/StylFormSign';
+import BtnSubmitFormSignIn from './../BtnsComponents/BtnSubmitFormSignIn';
+import emailImg from './../../img/icons/icon-email.svg';
+import passwordlImg from './../../img/icons/icon-lock.svg';
+import LinkPage from '../LinkPage';
+
+const FormSignUp = () => {
+  let [isEmail, setEmail] = React.useState(false);
+  let [isPassword, setPassword] = React.useState(false);
+
+  return (
+    <Formik
+      initialValues={{
+        email: '',
+        password: '',
+      }}
+      validate={(values) => {
+        const errors = {};
+        const regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/;
+        const regulEmail =
+          /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+        if (!regulPassword.test(values.password)) {
+          setPassword((isPassword = true));
+        } else {
+          setPassword((isPassword = false));
+        }
+
+        if (!regulEmail.test(values.email)) {
+          setEmail((isEmail = true));
+        } else {
+          setEmail((isEmail = false));
+        }
+
+        return errors;
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
```

Comment:
Something went wrong probably, setTimeout is the worst practice.

Reply:
`setTimeout` deleted